fiat/fiat-github/src/main/java/com/netflix/spinnaker/fiat/roles/github/client/GitHubAppAuthService.java

```diff
@@ -0,0 +1,179 @@
+    this.httpClient = httpClient;
+    this.objectMapper = new ObjectMapper();
+  }
+
+  public String getInstallationToken() throws IOException {
+    if (cachedToken == null || isTokenExpired(cachedToken)) {
+      synchronized (this) {
+        if (cachedToken == null || isTokenExpired(cachedToken)) {
+          cachedToken = fetchInstallationToken();
+        }
+      }
+    }
+    return cachedToken.getToken();
+  }
+
+  private InstallationToken fetchInstallationToken() throws IOException {
+    String jwt = generateJWT();
+
+    String url = baseUrl + "app/installations/" + installationId + "/access_tokens";
+    RequestBody body = RequestBody.create("{}", JSON);
+
+    Request request =
+        new Request.Builder()
+            .url(url)
+            .post(body)
+            .addHeader("Authorization", "Bearer " + jwt)
+            .addHeader("Accept", "application/vnd.github.v3+json")
+            .addHeader("User-Agent", "Spinnaker-Fiat")
+            .build();
+
+    try (Response response = httpClient.newCall(request).execute()) {
+      if (!response.isSuccessful()) {
+        String errorBody = response.body() != null ? response.body().string() : "No response body";
+        throw new IOException(
+            "Failed to get installation token. Status: "
+                + response.code()
+                + ", Body: "
+                + errorBody);
+      }
+
+      String responseBody = response.body().string();
+      InstallationToken token = objectMapper.readValue(responseBody, InstallationToken.class);
+
+      log.debug(
+          "Successfully obtained GitHub App installation token, expires at: {}",
+          token.getExpiresAt());
+      return token;
+    }
+  }
+
+  private String generateJWT() {
+    Instant now = Instant.now();
+    Date issuedAt = Date.from(now);
+    Date expiresAt = Date.from(now.plusSeconds(TimeUnit.MINUTES.toSeconds(JWT_EXPIRATION_MINUTES)));
+
+    return Jwts.builder()
+        .setIssuer(appId)
+        .setIssuedAt(issuedAt)
+        .setExpiration(expiresAt)
+        .signWith(privateKey, SignatureAlgorithm.RS256)
+        .compact();
+  }
+
+  private boolean isTokenExpired(InstallationToken token) {
+    if (token == null || !StringUtils.hasText(token.getExpiresAt())) {
+      return true;
+    }
+
+    try {
+      Instant expiresAt = Instant.parse(token.getExpiresAt());
+      Instant refreshThreshold =
+          Instant.now().plusSeconds(TimeUnit.MINUTES.toSeconds(TOKEN_REFRESH_BUFFER_MINUTES));
+      return expiresAt.isBefore(refreshThreshold);
+    } catch (Exception e) {
+      log.warn("Failed to parse token expiration time: {}", token.getExpiresAt(), e);
+      return true;
+    }
+  }
+
+  private PrivateKey loadPrivateKey(String privateKeyPath) {
+    try {
+      String content = new String(Files.readAllBytes(Paths.get(privateKeyPath)));
+
+      // Remove PEM headers and footers, and whitespace
+      String privateKeyPEM =
```

Comment:
Technically this should work, but... there's a few other alternative libraries that make loading private keys in PEM format much simpler/easier.  Look up bouncy castle - a quick google search gives some decent code via AI generation on using it's PemParser class.  There's some alternatives as well in Spring (aka https://docs.spring.io/spring-boot/3.3/api/java/org/springframework/boot/ssl/pem/PemContent.html which PRETTY sure exists in older versions).

Reply:
Handled in https://github.com/spinnaker/spinnaker/pull/7337/commits/2018566cba47b4326864cae462c405bb6e751bdf